### PR TITLE
GH-1295: Forgot password e2e tests.

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    });
+    // After requesting a reset password email, we should be taken back to the login page
+    cy.url().should('match', /\/login/);
+    // ... and display a success message.
+    cy.get('cx-global-message .alert-info').should(
+      'contain',
+      'An email has been sent to you with information on how to reset your password.'
+    );
+  });
+
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts]: Can we add test of "if click cancel, back to login page"?

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+context('Forgot Password Page', () => {
+  before(() => {
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts]: before go to forgot-password page, we need clear session to make sure there is really no user token exist (no user login).
cy.window().then(win => win.sessionStorage.clear());

Comment (plabadie) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts]: Fixed.  Good idea.  I didn't know we could do that.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/reset-password/forgot-password.e2e-spec.ts]: Absolutely.  Fixed. Good catch.

**Pull request merged**